domain/src/test/java/uk/gov/hmcts/ccd/definition/store/domain/validation/casefield/CaseFieldEntityComplexFieldACLValidatorImplTest.java

```diff
@@ -0,0 +1,274 @@
+    public void setup() {
+        MockitoAnnotations.initMocks(this);
+
+        userRole = new UserRoleEntity();
+        userRole.setReference(ROLE1);
+
+        complexACL0 = new ComplexFieldACLEntity();
+        complexACL0.setListElementCode("Class");
+        complexACL1 = new ComplexFieldACLEntity();
+        complexACL1.setListElementCode("Class.ClassMembers");
+        complexACL2 = new ComplexFieldACLEntity();
+        complexACL2.setListElementCode("Class.ClassMembers.Children");
+        complexACL3 = new ComplexFieldACLEntity();
+        complexACL3.setListElementCode("Class.ClassMembers.Children.ChildFullName");
+
+        given(caseFieldEntityValidationContext.getCaseReference()).willReturn("case_type");
+
+        caseField = new CaseFieldEntity();
+        caseField.setReference("case_field");
+
+        fieldACL = new CaseFieldACLEntity();
+
+        validator = new CaseFieldEntityComplexFieldACLValidatorImpl();
+    }
+
+    @Test
+    @DisplayName("should fail when user not found")
+    public void shouldHaveValidationErrorWhenUserNotFound() {
+        complexACL0.setUserRole(null);
+        caseField.addComplexFieldACL(complexACL0);
+        fieldACL.setUserRole(userRole);
+        caseField.addCaseFieldACL(fieldACL);
+
+        final ValidationResult result = validator.validate(caseField, caseFieldEntityValidationContext);
+
+        assertAll(
+            () -> assertThat(result.getValidationErrors().size(), is(2)),
+            () -> assertThat(result.getValidationErrors().get(0), instanceOf(CaseFieldEntityInvalidUserRoleValidationError.class)),
+            () -> assertThat(result.getValidationErrors().get(0).getDefaultMessage(), is(
+            "Invalid UserRole for case type 'case_type', case field 'case_field'"))
+        );
+    }
+
+    @Test
+    @DisplayName("should not fail when user is found")
+    public void shouldHaveNoValidationErrorWhenUserFound() {
+        complexACL0.setUserRole(userRole);
+        complexACL0.setCreate(true);
+        caseField.addComplexFieldACL(complexACL0);
+        fieldACL.setCreate(true);
+        fieldACL.setUserRole(userRole);
+        caseField.addCaseFieldACL(fieldACL);
+
+        final ValidationResult result = validator.validate(caseField, caseFieldEntityValidationContext);
+
+        assertThat(result.getValidationErrors(), empty());
+    }
+
+    @Test
+    @DisplayName("should fail when parent caseField has no ACL defined")
+    public void shouldSucceedWhenComplexChildHasLessOrEqualAccessComparedToCaseFieldParent() {
+        complexACL0.setUserRole(userRole);
+        complexACL0.setCreate(true);
+        caseField.addComplexFieldACL(complexACL0);
+        caseField.addCaseFieldACL(fieldACL);
+
+        final ValidationResult result = validator.validate(caseField, caseFieldEntityValidationContext);
+
+        assertThat(result.getValidationErrors(), not(empty()));
+    }
+
+    @Test
+    @DisplayName("should fail when parent caseField has no ACL defined")
```

Comment:
Same DisplayName to the test above?

Reply:
Looks like the same test as above but one more assert. Can there be one instead?

Reply:
👍